src/libcharon/plugins/vici/python/vici/protocol.py

```diff
@@ -34,6 +34,8 @@ def _recvall(self, count):
         data = b""
         while len(data) < count:
             data += self.socket.recv(count - len(data))
+            if not data:
```

Comment:
Checking this after a `+=` operation seems not ideal. It's possible that a small amount was read before the socket was closed resulting in the same spinning you intended to prevent.